src/ovp_pipeline/digest_inputs.py

```
@@ -0,0 +1,1019 @@
+        connected_community_crystals=tuple(connected_crystals),
+        touched_contradictions=tuple(touched_contradictions),
+        recent_top_crystals=tuple(recent_top),
+    )
+
+
+# ---------------------------------------------------------------
+# Layer 3 — pipeline state
+# ---------------------------------------------------------------
+
+
+def _collect_layer3(
```

Ruff (0.15.12)

[warning] Too many branches (16 > 12)

(PLR0912)